oak-doc/src/site/markdown/security/authorization/bestpractices.md

```diff
@@ -0,0 +1,270 @@
+to edit the permission setup of your Oak installation. This will help you avoid common pitfalls. If you find yourself 
+granting your _content-writers_ role full access to just get it work, you probably left your application vulnerable.
+
+- JCR Specification sections [Access Control Management](https://s.apache.org/jcr-2.0-spec/16_Access_Control_Management.html) 
+and [Permissions and Capabilities](https://s.apache.org/jcr-2.0-spec/9_Permissions_and_Capabilities.html)
+- [Oak Authorization Documentation](../authorization.html) with separate sections for [Access Control Management](../accesscontrol.html) and [Permission Evaluation](../permission.html).
+- Exercises for authorization topics below https://github.com/apache/jackrabbit-oak/tree/trunk/oak-exercise/src/test/java/org/apache/jackrabbit/oak/exercise/security/authorization
+
+### Principle of least privilege
+
+Keep in mind that not having any permissions granted is equivalent to denying everything (which is in 
+this case redundant). Start without any access and then keep granting permissions as needed, following the 
+[principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege).
+In other words: only grant the minimal set of privileges required to perform a particular task.
+
+### Verification
+
+Write tests upfront that verify for each role and task the expected effective permissions (see definition of roles) are 
+granted. Neither less nor more.
+
+Ideally, your tests will fail as soon as someone is attempting to make any change to the permission setup.
+Granting additional permissions may open up the door for a privilege escalation and revoking permissions will break
+your application (if it doesn't you didn't follow the principle of least privilege).
+
+This may also include tests verify that really no permissions are granted at resources that are outside the scope of a 
+given role/task
+
+## Oak Specific Best Practices
+
+### Avoid deny
+
+All authorization models present with Apache Jackrabbbit Oak start without any access granted by default i.e. 
+implicit deny everywhere. It is therefore recommended to only grant access where needed and avoid adding explicit 
+deny access control entries. In particular in combination with subsequent allows the overall effect will be hard to 
+understand as soon as multiple principals are contained in a given subject.
+
+Be wary if you find yourself adding combinations of denies and allows as it might highlight problematic patterns in 
+your content model that will be hard to secure over time.
+
+### Avoid redundancy
+
+Don't specify redundant access control setup just to be on the safe side:
+
+- If access is granted, avoid repeating the same setup down the hierarchy.
+- Avoid setup for principals with administrative access for which permission evaluation is omitted. It might even create a false sense of security.
```

Comment:
can you either elaborate what ```principals with administrative access``` are or add a link to such a description?

Reply:
sure. i will try to clarify and link to the corresponding configuration option